core/src/test/java/org/apache/calcite/test/enumerable/EnumerableRecursiveUnionTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.test.enumerable;
+
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.test.CalciteAssert;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+
+/**
+ * Unit tests for
+ * {@link org.apache.calcite.adapter.enumerable.EnumerableRecursiveUnion}
+ * and {@link org.apache.calcite.adapter.enumerable.EnumerableDeltaTableScan}
+ * <a href="https://issues.apache.org/jira/browse/CALCITE-2812">[CALCITE-2812]
+ * Add algebraic operators to allow expressing recursive queries</a>.
+ */
+public class EnumerableRecursiveUnionTest {
```

Comment:
@rubenada can you add also a few tests where the recursive part is a bit more complicated (for instance a join between a delta scan and a regular table)?

Reply:
New unit tests have been added